python/cudf/cudf/core/column/column.py

```diff
@@ -80,6 +81,7 @@
 from cudf.utils.utils import NotIterable, mask_dtype
 
 T = TypeVar("T", bound="ColumnBase")
+Slice = TypeVar("Slice", bound=slice)  # WAR for "slice" is taken in ColumnBase
```

Comment:
I want to type annotate `key` with type `slice`. But `ColumnBase` already has a function named `slice` bind to it. So I created an alias to bind to `slice` before `ColumnBase` binds it.

Reply:
On a second thought, we probably shouldn't create any objects that shadows the internal function names. Should we rename `ColumnBase.slice` to `ColumnBase.slice_` instead?

Reply:
I realized this was an issue when I first reviewed but I didn't suggest changing it because it's out of scope for this PR. I would say come back and address it later.

Reply:
Okay, I understand now. I'd file an issue and do this in a follow-up PR.

Reply:
```suggestion
# TODO: This workaround allows type hints for `slice`, since `slice` is a method in ColumnBase.
Slice = TypeVar("Slice", bound=slice)
```